### PR TITLE
Fix 'rtol' and 'atol' for numeric extension types

### PR DESCRIPTION
**Pull request by jonashaag**
- [ ] closes #xxxx
- [x] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

cc @stefansorgqc

### CODE REVIEW

File: pandas/tests/util/test_assert_series_equal.py [hunk 1]
```diff
@@ -115,7 +117,10 @@ def test_less_precise(data1, data2, dtype, decimals):
     if (decimals == 5 or decimals == 10) or (
         decimals >= 3 and abs(data1 - data2) >= 0.0005
     ):
-        msg = "Series values are different"
+        if is_extension_array_dtype(dtype):
```
Comment (jreback) [inline pandas/tests/util/test_assert_series_equal.py:120]: why would this message change? the Series are still different

**Review by jreback — CHANGES_REQUESTED** [review]
does this have an associated issue / bug report?

File: pandas/tests/util/test_assert_series_equal.py (hunk 1 above)
Comment (jonashaag) [reply pandas/tests/util/test_assert_series_equal.py:120]:
```
In [6]: pd.testing.assert_series_equal(pd.Series([1]), pd.Series([2]))
...

AssertionError: Series are different

Series values are different (100.0 %)
[index]: [0]
[left]:  [1]
[right]: [2]
```

```
In [7]: pd.testing.assert_series_equal(pd.Series([1], dtype="Int64"), pd.Series(
   ...: [2], dtype="Int64"))
...

AssertionError: ExtensionArray are different

ExtensionArray values are different (100.0 %)
[index]: [0]
[left]:  [1]
[right]: [2]
```

**Comment by jonashaag** [comment]
> does this have an associated issue / bug report?

No, shall I create one?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jonashaag

**Pull request merged**